packages/react-client/src/ReactFlightClient.js

```diff
@@ -902,6 +902,20 @@ function waitForReference<T>(
       handler.value = parentObject[key];
     }
 
+    // If the parent object is an unparsed React element tuple and its outlined
+    // props have now been resolved, we also need to update the props of the
+    // parsed element object (i.e. handler.value).
+    if (
+      parentObject[0] === REACT_ELEMENT_TYPE &&
+      key === '3' &&
+      typeof handler.value === 'object' &&
+      handler.value !== null &&
+      handler.value.$$typeof === REACT_ELEMENT_TYPE &&
+      handler.value.props === null
```

Comment:
Are there cases where the props are already resolved i.e.  `handler.value.props !== null`?

Reply:
I'm not sure. At least there's no test failing without the check. But since we don't have 100% test coverage in the flight client, I wanted to be on the safe side.

Reply:
When we do this at the root it's actually a feature because we don't know that the empty `''` key means that it's at the root. The `null` check implies that we're at the root because we couldn't be waiting for anything else until the root is done.

In this case, this check is kind of unnecessary. If you trust that we do the rest of the logic correct so you don't need it, and if we don't do it right it'll probably break anyway.

So I think just remove this since you need to be able to do this for every other key anyway (key, props, owner, stack...). It's just unnecessary to check all of them.